lib/kennel/utils.rb

```diff
@@ -105,18 +105,22 @@ def parallel(items, max: 10)
         threads = [items.size, max].min
         work = items.each_with_index.to_a
         done = Array.new(items.size)
+        mutex = Mutex.new
         workers = Array.new(threads).map do
           Thread.new do
             loop do
-              item, i = work.pop
+              item, i = mutex.synchronize { work.pop }
```

Comment:
pop is thread-safe

Reply:
Citation? 

I've just dug around a bit in the ruby documentation and failed to find anything backing up your assertions about things being thread safe.